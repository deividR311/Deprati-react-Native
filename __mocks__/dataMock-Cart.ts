export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
