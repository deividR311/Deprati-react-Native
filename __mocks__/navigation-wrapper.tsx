import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
