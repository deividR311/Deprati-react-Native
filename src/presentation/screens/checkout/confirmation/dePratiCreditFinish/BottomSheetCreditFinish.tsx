import React, { useEffect, useState } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { useRoute } from '@react-navigation/native';
import { useFormik } from 'formik';
import {
  COLORS,
  FontStyles,
  FONTS_FAMILY
} from '../../../../../application/common';
import useContentServiceRequest from '../../../../../infrastucture/apis/contentService/useContentServiceRequest.hook';
import {
  BottomSheet,
  IconButton
} from '../../../../common-components/bottomSheet';
import Button, {
  ButtonText
} from '../../../../common-components/buttons/Button';
import ImageCard from '../../../../common-components/credito/ImageCard';
import InputBase from '../../../../common-components/inputs/InputBase';
import ErrorText from '../../../../common-components/texts/ErrorText';
import { Props } from './bottomsheet.interface';
import { SkeletonCreditFinish } from './SkeletonCreditFinish';
import { useAppSelector } from '../../../../../application/state-manager';
import { CreditInfoSelector } from '../../../../../application/state-manager/services/checkout';

const WE_HAVE_SENT_TEXT =
  'Hemos enviado a tu celular\n[PHONE] [EMAIL] tu código de compra para que continúes con la transacción.';
const FOOTER_TEXT =
  'Si no has recibido el código de compra o no has actualizado tu correo electrónico, llama al (04) 3731800 o (02) 3950400. También puedes acercarte a las áreas de Servicio al Cliente en nuestras tiendas.';
const VALID_CODE_TEXT = 'Código válido por 5 minutos';
const VALIDATE = 'VALIDAR Y PAGAR';
const RESEND_CODE_TEXT = 'Reenviar código de compra';

export default function BottomSheetCreditFinish({
  onCloseRequest,
  show,
  hook
}: Props) {
  const route = useRoute();
  const { params } = route;
  //@ts-ignore
  const { maskedPhoneNumber, maskedEmail } = params;
  const [disabledResend] = useState(false);
  const creditInfo = useAppSelector(CreditInfoSelector);
  const { getContentCheckout, dataChekout, isLoadingContentService } =
    useContentServiceRequest();

  const formik = useFormik({
    ...FORMIK_SETTINGS,
    validateOnMount: false,
    validateOnChange: false,
    validateOnBlur: false,
    onSubmit: values => doSubmit(values.code)
  });

  const doSubmit = (codeInput: string) => {
    hook.setMsgErrorCodeDePratiCredit('');
    hook.setDirectCreditAuthorize(codeInput);
  };

  const handleResend = () => {
    hook.getDirectCreditToken();
  };

  const handleContentCard = (): string => {
    //console.log('handleContentCard', route)
    let text_show = WE_HAVE_SENT_TEXT;

    text_show = text_show.replace(
      '[PHONE]',
      `${maskedPhoneNumber || creditInfo?.maskedPhoneNumber || ''}`
    );

    text_show = text_show.replace(
      ' [EMAIL]',
      maskedEmail
        ? ` y a tu mail \n${maskedEmail || creditInfo?.maskedEmail || ''}`
        : ''
    );

    return text_show;
  };

  useEffect(() => {
    if (show) {
      getContentCheckout();
    }
  }, [show]);

  const handleClose = () => {
    formik.resetForm();
    hook.setMsgErrorCodeDePratiCredit('');
    hook.setShowBottomSheetDePratiCredit(false);
    onCloseRequest();
  };

  return (
    <BottomSheet
      percentage={100}
      show={show}
      canDrop={false}
      onCloseRequest={handleClose}
      header={
        <View style={styles.close}>
          <IconButton
            testID="close_modal"
            iconName="close"
            onPress={handleClose}
          />
        </View>
      }
      paddingHorizontal={24}>
      <ImageCard style={{ width: 230, height: 236, marginTop: -8 }} />

      <View style={styles.card_container}>
        <Text style={styles.card_title}>{handleContentCard()}</Text>
        <Text style={styles.card_subtitle}>{VALID_CODE_TEXT}</Text>

        <InputBase
          style={styles.input_code}
          value={formik.values.code}
          onChangeText={(text: string) => formik.setFieldValue('code', text)}
          placeholder="Ingresa el código aquí"
          keyboardType="number-pad"
          label=""
          maxLength={10}
          dense
        />
        {formik.errors.code && <ErrorText text={formik.errors.code} />}
        {hook?.msgErrorCodeDePratiCredit?.length > 1 && (
          <ErrorText text={hook.msgErrorCodeDePratiCredit} />
        )}
        <Button
          disabled={
            formik.values.code.length < 5 ||
            hook.isLoadingAuthorize ||
            hook.isLoadingGetToken
          }
          marginTop={16}
          linkName={VALIDATE}
          containerStyle={styles.button}
          showActivityIndicator={
            hook.isLoadingAuthorize || hook.isLoadingGetToken
          }
          activityIndicator={{
            color: FontStyles.LightColor.color
          }}
          onPress={formik.submitForm}
          backgroundColor={FontStyles.PrimaryColor.color}
          textColor={FontStyles.LightColor.color}
        />

        <ButtonText
          disabled={disabledResend}
          onPress={handleResend}
          styleText={{ color: FontStyles.PrimaryColor.color }}
          style={{ marginVertical: 28 }}
          title={RESEND_CODE_TEXT}
        />
      </View>
      {isLoadingContentService ? (
        <SkeletonCreditFinish />
      ) : (
        <Text style={styles.footer_text}>
          {dataChekout?.checkoutOrderConfirmation?.directCredit?.legend ??
            FOOTER_TEXT}
        </Text>
      )}
    </BottomSheet>
  );
}

const FORMIK_SETTINGS = {
  initialValues: {
    code: ''
  }
};

const styles = StyleSheet.create({
  close: {
    alignItems: 'flex-end',
    justifyContent: 'flex-end',
    paddingBottom: 11,
    width: '100%'
  },
  card_container: {
    backgroundColor: COLORS.GRAYDARK20,
    paddingHorizontal: 18,
    borderRadius: 6,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1
    },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,

    elevation: 3
  },
  card_title: { ...FontStyles.Body_1, textAlign: 'center', marginTop: 16 },
  card_subtitle: {
    ...FontStyles.Body_2,
    marginVertical: 16,
    textAlign: 'center'
  },
  input_code: {
    backgroundColor: COLORS.GRAYDARK20,
    height: 47
  },
  button: { height: 47 },
  footer_text: {
    ...FontStyles.Subtitle_1,
    textAlign: 'center',
    marginTop: 24,
    fontFamily: FONTS_FAMILY.Roboto,
    paddingBottom: 20
  }
});
