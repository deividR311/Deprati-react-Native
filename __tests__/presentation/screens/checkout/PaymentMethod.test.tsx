import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import {
  confirmation,
  sumaryPurchase
} from '../../../../__mocks__/dataMock-ChoosePaymentMethod';
import BottomSheetCreditFinish from '../../../../src/presentation/screens/checkout/confirmation/dePratiCreditFinish/BottomSheetCreditFinish';
import SummaryPurchase from '../../../../src/presentation/screens/checkout/summaryPurchase/summaryPurchase';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn()
    }),
    useRoute: () => ({
      params: {
        maskedPhoneNumber: '991XXX404',
        maskedEmail: ''
      }
    })
  };
});

const ReturnScreen = (
  <BottomSheetCreditFinish
    onCloseRequest={() => jest.fn()}
    show={confirmation.showBottomSheetDePratiCredit}
    hook={confirmation}
  />
);

let screenTest: any;
describe('BottomSheetCreditFinish', () => {
  beforeEach(() => {
    screenTest = render(<NavigationWrapper>{ReturnScreen}</NavigationWrapper>);
  });

  it('renders correctly in Snapshot', () => {
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });

  it('Validate Msg Code', async () => {
    const { getByText } = screenTest;
    expect(getByText('VALIDAR Y PAGAR')).toBeDefined();
  });

  it('open accordion on first render', async () => {
    const { queryAllByTestId } = render(
      <SummaryPurchase
        hookConfirmation={{ ...confirmation, firstTimeSummary: true }}
        hookSummaryPurchase={sumaryPurchase}
      />
    );
    expect(queryAllByTestId('accordion-expanded')).toHaveLength(2);
  });

  it('open accordion on after first render', async () => {
    const { queryAllByTestId } = render(
      <SummaryPurchase
        hookConfirmation={{ ...confirmation, firstTimeSummary: false }}
        hookSummaryPurchase={sumaryPurchase}
      />
    );
    expect(queryAllByTestId('accordion-expanded')).toHaveLength(0);
  });
});
