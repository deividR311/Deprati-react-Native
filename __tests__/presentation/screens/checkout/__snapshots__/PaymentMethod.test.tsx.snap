// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomSheetCreditFinish renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <Modal
          animationType="fade"
          hardwareAccelerated={false}
          onDismiss={[Function]}
          onRequestClose={[Function]}
          transparent={true}
          visible={true}
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.2)",
                "flex": 1,
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />
            </View>
            <View
              onLayout={[Function]}
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderTopLeftRadius": 13,
                    "borderTopRightRadius": 13,
                    "elevation": 24,
                    "height": 1334,
                    "overflow": "hidden",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 12,
                      "width": 0,
                    },
                    "shadowOpacity": 0.58,
                    "shadowRadius": 16,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "justifyContent": "flex-end",
                      "paddingBottom": 11,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    accessibilityLabel="close_modal"
                    accessibilityRole="button"
                    accessibilityState={Object {}}
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#E6E6E6",
                        "borderRadius": 4,
                        "height": 28,
                        "justifyContent": "center",
                        "opacity": 1,
                        "width": 28,
                      }
                    }
                    testID="close_modal"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 18,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <RCTScrollView
                  contentContainerStyle={Object {}}
                  style={
                    Array [
                      Object {
                        "paddingHorizontal": 24,
                      },
                      Object {},
                    ]
                  }
                >
                  <View>
                    <View
                      accessibilityIgnoresInvertColors={true}
                      style={
                        Array [
                          Object {
                            "alignSelf": "center",
                            "height": 143,
                            "justifyContent": "center",
                            "marginTop": 32,
                            "width": 235,
                          },
                          Object {
                            "height": 236,
                            "marginTop": -8,
                            "width": 230,
                          },
                        ]
                      }
                    >
                      <Image
                        resizeMode="contain"
                        source={
                          Object {
                            "testUri": "../../../assets/images/cardCredito.png",
                          }
                        }
                        style={
                          Array [
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            Object {
                              "height": 236,
                              "width": 230,
                            },
                            undefined,
                          ]
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#F5F6F9",
                          "borderRadius": 6,
                          "elevation": 3,
                          "paddingHorizontal": 18,
                          "shadowColor": "#000",
                          "shadowOffset": Object {
                            "height": 1,
                            "width": 0,
                          },
                          "shadowOpacity": 0.22,
                          "shadowRadius": 2.22,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#000000",
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 16,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0.44,
                            "lineHeight": 24,
                            "marginTop": 16,
                            "textAlign": "center",
                          }
                        }
                      >
                        Hemos enviado a tu celular
991XXX404 tu código de compra para que continúes con la transacción.
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#000000",
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 14,
                            "fontWeight": "400",
                            "letterSpacing": 0.25,
                            "lineHeight": 20,
                            "marginVertical": 16,
                            "textAlign": "center",
                          }
                        }
                      >
                        Código válido por 5 minutos
                      </Text>
                      <View
                        style={Object {}}
                      >
                        <View
                          pointerEvents="none"
                          style={
                            Array [
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 6,
                              },
                              Object {
                                "backgroundColor": "#F5F6F9",
                                "borderColor": "#979797",
                                "borderRadius": 4,
                                "borderWidth": 1,
                              },
                            ]
                          }
                          testID="text-input-outline"
                        />
                        <View>
                          <View
                            style={
                              Array [
                                Object {
                                  "paddingBottom": 0,
                                },
                                Object {
                                  "minHeight": 47,
                                  "paddingTop": 8,
                                },
                              ]
                            }
                          >
                            <TextInput
                              accessibilityIgnoresInvertColors={true}
                              autoComplete="off"
                              editable={true}
                              keyboardType="number-pad"
                              maxFontSizeMultiplier={1.5}
                              maxLength={10}
                              multiline={false}
                              onBlur={[Function]}
                              onChangeText={[Function]}
                              onFocus={[Function]}
                              placeholder="Ingresa el código aquí"
                              placeholderTextColor="#979797"
                              selectionColor="#979797"
                              style={
                                Array [
                                  Object {
                                    "flexGrow": 1,
                                    "margin": 0,
                                    "paddingHorizontal": 14,
                                    "zIndex": 1,
                                  },
                                  Object {
                                    "height": 47,
                                  },
                                  Object {
                                    "paddingBottom": 15.5,
                                    "paddingTop": 15.5,
                                  },
                                  Object {
                                    "color": "#000000",
                                    "fontFamily": "System",
                                    "fontSize": 16,
                                    "fontWeight": undefined,
                                    "textAlign": "left",
                                    "textAlignVertical": "center",
                                  },
                                  false,
                                  Array [
                                    Object {},
                                  ],
                                ]
                              }
                              underlineColorAndroid="transparent"
                              value=""
                            />
                          </View>
                        </View>
                      </View>
                      <View
                        accessibilityLabel="VALIDAR Y PAGAR"
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#D3D3D3",
                            "borderRadius": 5,
                            "flexDirection": "row",
                            "height": 47,
                            "justifyContent": "center",
                            "marginTop": 16,
                            "opacity": 1,
                            "paddingVertical": 10,
                          }
                        }
                        testID="VALIDAR Y PAGAR"
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000000",
                                "fontFamily": "Roboto-Regular",
                                "fontSize": 14,
                                "fontWeight": "700",
                                "letterSpacing": 0.8,
                                "textAlign": "center",
                              },
                              undefined,
                              Object {
                                "color": "#ffffff",
                                "opacity": 0.5,
                              },
                            ]
                          }
                        >
                          VALIDAR Y PAGAR
                        </Text>
                      </View>
                      <View
                        accessibilityLabel=""
                        accessibilityState={
                          Object {
                            "disabled": false,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                            "marginVertical": 28,
                            "opacity": 1,
                            "paddingVertical": 14,
                          }
                        }
                        testID=""
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#263238",
                                "fontFamily": "Roboto-Regular",
                                "fontSize": 14,
                                "fontWeight": undefined,
                                "letterSpacing": 0.6,
                                "textDecorationLine": "underline",
                              },
                              Object {
                                "color": "#E31A2C",
                              },
                              Object {},
                            ]
                          }
                        >
                          Reenviar código de compra
                        </Text>
                      </View>
                    </View>
                    <View
                      onLayout={[Function]}
                      style={
                        Object {
                          "marginVertical": 8,
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignSelf": "center",
                            "backgroundColor": "#E1E9EE",
                            "borderRadius": 4,
                            "height": 70,
                            "key": "creditfinish",
                            "overflow": "hidden",
                            "width": "95%",
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "height": "100%",
                                "position": "absolute",
                                "width": "100%",
                              },
                              Object {
                                "transform": Array [
                                  Object {
                                    "translateX": undefined,
                                  },
                                ],
                              },
                            ]
                          }
                        >
                          <BVLinearGradient
                            colors={
                              Array [
                                4292995566,
                                4294113532,
                                4292995566,
                              ]
                            }
                            endPoint={
                              Object {
                                "x": 1,
                                "y": 0,
                              }
                            }
                            locations={null}
                            startPoint={
                              Object {
                                "x": 0,
                                "y": 0,
                              }
                            }
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
